orc8r/protos/tenants.proto

```diff
@@ -35,10 +35,17 @@ message IDAndTenant {
     Tenant tenant = 2;
 }
 
+message IDAndControlProxy {
+  int64 id = 1;
+  string controlProxy = 2;
+}
+
 service TenantsService {
     rpc GetAllTenants (Void) returns (TenantList) {}
     rpc GetTenant (GetTenantRequest) returns (Tenant) {}
     rpc CreateTenant (IDAndTenant) returns (Void) {}
     rpc SetTenant (IDAndTenant) returns (Void) {}
     rpc DeleteTenant (GetTenantRequest) returns (Void) {}
+    rpc GetControlProxy (GetTenantRequest) returns (IDAndControlProxy) {}
+    rpc CreateOrUpdateControlProxy (IDAndControlProxy) returns (Void) {}
```

Comment:
We always want the req and res to be `<rpc name>Request` and `<rpc name>Response` -- I know we don't follow that pattern correctly everywhere 🙂

Reply:
Will do, but also just wondering why? Since the message is the same thing duplicated

Reply:
For sure. So the reason is backward compatibility -- *currently* they are the same message (same fields), but that might change over time

Specifically, to preserve proto/grpc backward compatibility, there are a specific set of rules, but basically, if you only append new fields, you're in the clear

Potential issues of sharing, especially when a large number of rpcs share 1 underlying msg type

1. RPCs could diverge in purpose over time, which would cause the shared message type to effectively multiplex multiple logical message types, which causes confusion and larger surface for bugs
2. We could need to deprecate a field, but just for 1 of the services -- no clear solution in that case, so it just wouldn't get deprecated

More info: https://github.com/uber/prototool/blob/dev/style/README.md#servicesrpcs

Reply:
Obv this isn't really an issue for our orc8r-internal grpc/proto defns, but it's good practice to keep this consistent across all gprc/proto defns

Reply:
Thanks for all the info!